File: src/org/onion_lang/onion/compiler/util/SymbolGenerator.java
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2005, Kota Mizushima, All rights reserved.       *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package org.onion_lang.onion.compiler.util;

/**
 * @author Kota Mizushima
 * Date: 2005/07/07
 */
public class SymbolGenerator {
  private final String prefix;
  private int count;
  
  public SymbolGenerator(String prefix) {
    this.prefix = prefix;
  }

  public String getPrefix(){
    return prefix;
  }
  
  public int getCount(){
    return count;
  }
  
  public String generate(){
    String newSymbol = prefix + count;
    count++;
    return newSymbol;
  }
}

File: src/main/scala/onion/compiler/toolbox/SymbolGenerator.scala
/* ************************************************************** *
 *                                                                *
 * Copyright (c) 2005-2012, Kota Mizushima, All rights reserved.  *
 *                                                                *
 *                                                                *
 * This software is distributed under the modified BSD License.   *
 * ************************************************************** */
package onion.compiler.toolbox

/**
 * @author Kota Mizushima
 *
 */
class SymbolGenerator(val prefix: String) {
  private var count_ : Int = 0

  def count: Int = count_

  def generate: String = {
    val newSymbol: String = prefix + count_
    count_ += 1
    newSymbol
  }
}